src/lib/support/ThreadOperationalDataset.cpp

```diff
@@ -0,0 +1,525 @@
+
+    /**
+     * This method returns the Tlv type.
+     *
+     * @returns The Tlv type.
+     *
+     */
+    uint8_t GetType(void) const { return mType; }
+
+    /**
+     * This method sets the Tlv type.
+     *
+     */
+    void SetType(uint8_t aType) { mType = aType; }
+
+    /**
+     * This method returns the Tlv length.
+     *
+     * @returns The Tlv length.
+     *
+     */
+    uint8_t GetLength(void) const { return mLength; }
+
+    /**
+     * This method sets the length.
+     */
+    void SetLength(uint8_t aLength) { mLength = aLength; }
+
+    /**
+     * This method returns a pointer to the value.
+     *
+     * @returns The Tlv value.
+     *
+     */
+    const void * GetValue(void) const
+    {
+        return reinterpret_cast<const uint8_t *>(this) + sizeof(mType) +
+            (mLength != kLengthEscape ? sizeof(mLength) : (sizeof(uint16_t) + sizeof(mLength)));
+    }
+
+    void * GetValue(void) { return const_cast<void *>(const_cast<const Tlv *>(this)->GetValue()); }
+
+    /**
+     * This method returns the value as a uint16_t.
+     *
+     * @returns The uint16_t value.
+     *
+     */
+    uint16_t GetValueUInt16(void) const
+    {
+        const uint8_t * p = static_cast<const uint8_t *>(GetValue());
+
+        return static_cast<uint16_t>(p[0] << 8 | p[1]);
+    }
+
+    /**
+     * This method returns the value as a uint8_t.
+     *
+     * @returns The uint8_t value.
+     *
+     */
+    uint8_t GetValueUInt8(void) const { return *static_cast<const uint8_t *>(GetValue()); }
+
+    /**
+     * This method sets uint16_t as the value.
+     *
+     * @param[in]    aValue         uint16_t value
+     *
+     */
+    void SetValue(uint16_t aValue)
+    {
+        uint8_t * value;
+
+        SetLength(sizeof(aValue));
+        value    = static_cast<uint8_t *>(GetValue());
+        value[0] = static_cast<uint8_t>(aValue >> 8);
+        value[1] = static_cast<uint8_t>(aValue & 0xff);
+    }
+
+    /**
+     * This method sets uint8_t as the value.
+     *
+     * @param[in]    aValue         uint8_t value
+     *
+     */
+    void SetValue(uint8_t aValue)
+    {
+        SetLength(sizeof(aValue));
+        *static_cast<uint8_t *>(GetValue()) = aValue;
+    }
+
+    /**
+     * This method sets int8_t as the value.
+     *
+     * @param[in]    aValue         int8_t value
+     *
+     */
+    void SetValue(int8_t aValue)
+    {
+        SetLength(sizeof(aValue));
+        *static_cast<int8_t *>(GetValue()) = aValue;
+    }
+
+    /**
+     * This method copies the value.
+     */
+    void SetValue(const void * aValue, uint8_t aLength)
+    {
+        SetLength(aLength);
+        memcpy(GetValue(), aValue, aLength);
+    }
+
+    /**
+     * This method returns the pointer to the next Tlv.
+     *
+     * @returns A pointer to the next Tlv.
+     *
+     */
+    const Tlv * GetNext(void) const
+    {
+        return reinterpret_cast<const Tlv *>(static_cast<const uint8_t *>(GetValue()) + GetLength());
```

Comment:
Shouldn't this method handle `kLengthEscape` case?

Reply:
The maximum of Thread operational dataset is limited to 254.

Reply:
Ok, but `GetValue` method handles it, which is confusing. If it can never happen I would suggest removing `kLengthEscape` completely, otherwise we must handle it consistently in all related methods.

Reply:
added assertions.